diff -Naur freetype-2.8.1/include/freetype/config/ftoption.h freetype-2.8.1-new/include/freetype/config/ftoption.h
--- freetype-2.8.1/include/freetype/config/ftoption.h	2017-08-23 12:55:39.000000000 +0800
+++ freetype-2.8.1-new/include/freetype/config/ftoption.h	2017-11-10 10:47:12.033837914 +0800
@@ -119,7 +119,7 @@
   /* rendering technology that produces excellent output without LCD       */
   /* filtering.                                                            */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /*************************************************************************/
