diff -Naur libxcb-1.12/configure.ac libxcb-1.12-new/configure.ac
--- libxcb-1.12/configure.ac	2016-05-14 16:33:49.000000000 +0800
+++ libxcb-1.12-new/configure.ac	2017-11-20 15:04:35.919166100 +0800
@@ -72,7 +72,7 @@
 
 # Find the xcb-proto protocol descriptions
 AC_MSG_CHECKING(XCBPROTO_XCBINCLUDEDIR)
-XCBPROTO_XCBINCLUDEDIR=`$PKG_CONFIG --variable=xcbincludedir xcb-proto`
+XCBPROTO_XCBINCLUDEDIR=$lt_sysroot`$PKG_CONFIG --variable=xcbincludedir xcb-proto`
 AC_MSG_RESULT($XCBPROTO_XCBINCLUDEDIR)
 AC_SUBST(XCBPROTO_XCBINCLUDEDIR)
 
@@ -82,7 +82,7 @@
 
 # Find the xcbgen Python package
 AC_MSG_CHECKING(XCBPROTO_XCBPYTHONDIR)
-XCBPROTO_XCBPYTHONDIR=`$PKG_CONFIG --variable=pythondir xcb-proto`
+XCBPROTO_XCBPYTHONDIR=$lt_sysroot`$PKG_CONFIG --variable=pythondir xcb-proto`
 AC_MSG_RESULT($XCBPROTO_XCBPYTHONDIR)
 AC_SUBST(XCBPROTO_XCBPYTHONDIR)
 
