diff -Naur kmod-20120815/configure.ac kmod-20120815-new/configure.ac
--- kmod-20120815/configure.ac	2012-08-15 10:29:19.000000000 +0800
+++ kmod-20120815-new/configure.ac	2012-12-15 16:48:02.891537490 +0800
@@ -108,10 +108,7 @@
 	AC_DEFINE(ENABLE_DEBUG, [1], [Debug messages.])
 ])
 
-m4_ifdef([GTK_DOC_CHECK], [
-GTK_DOC_CHECK([1.14],[--flavour no-tmpl-flat])
-], [
-AM_CONDITIONAL([ENABLE_GTK_DOC], false)])
+AM_CONDITIONAL([ENABLE_GTK_DOC], false)
 
 
 #####################################################################
