diff -Naur libXtst-1.2.1/xtst.pc.in libXtst-1.2.1-new//xtst.pc.in
--- libXtst-1.2.1/xtst.pc.in	2012-03-08 13:11:34.000000000 +0800
+++ libXtst-1.2.1-new//xtst.pc.in	2012-12-02 22:47:52.238020651 +0800
@@ -6,7 +6,7 @@
 Name: Xtst
 Description: The Xtst Library
 Version: @PACKAGE_VERSION@
-Requires: recordproto
-Requires.private: x11 xext
+Requires: recordproto x11 xext
+Requires.private: 
 Cflags: -I${includedir}
 Libs: -L${libdir} -lXtst
