diff -Naur linux-2.6.35.3-fsl-loco-v2/arch/arm/Makefile linux-2.6.35.3-fsl-loco-v2-new/arch/arm/Makefile
--- linux-2.6.35.3-fsl-loco-v2/arch/arm/Makefile	2012-11-05 13:21:42.000000000 +0800
+++ linux-2.6.35.3-fsl-loco-v2-new/arch/arm/Makefile	2012-11-21 08:37:18.369423188 +0800
@@ -93,6 +93,10 @@
 CFLAGS_ABI	+=-funwind-tables
 endif
 
+ifeq ($(CONFIG_ALIGNMENT_TRAP),y)
+CFLAGS_ABI  +=$(call cc-option,-mno-unaligned-access,)
+endif
+
 ifeq ($(CONFIG_THUMB2_KERNEL),y)
 AFLAGS_AUTOIT	:=$(call as-option,-Wa$(comma)-mimplicit-it=always,-Wa$(comma)-mauto-it)
 AFLAGS_NOWARN	:=$(call as-option,-Wa$(comma)-mno-warn-deprecated,-Wa$(comma)-W)
